src/schemes/boundary/open_boundary/method_of_characteristics.jl

```diff
@@ -165,8 +165,9 @@ function evaluate_characteristics!(system, v, u, v_ode, u_ode, semi, t)
     # of fluid particles by using the average of the values of the previous time step.
     # See eq. 27 in Negi (2020) https://doi.org/10.1016/j.cma.2020.113119
     @threaded semi for particle in each_integrated_particle(system)
-        # Particle is outside of the influence of fluid particles
-        if isapprox(volume[particle], 0)
+        # Particle is outside of the influence of fluid particles.
+        # Handle numerical precision issues (see also https://github.com/trixi-framework/TrixiParticles.jl/pull/913)
+        if volume[particle] * smoothing_length^ndims(system) < eps(eltype(smoothing_length))
```

Comment:
```suggestion
        # `volume` is in the order of 1 / h^d, so volume * h^d is in the order of 1.
        if volume[particle] * smoothing_length^ndims(system) < eps(eltype(smoothing_length))
```
It is also confusing that `volume` is not actually a volume but an inverse volume. Would it make sense to change the variable name in another PR?